plugins/processors/dcos_metadata/dcos_metadata.go

```diff
@@ -296,16 +299,26 @@ func mapExecutorNames(ge *agent.Response_GetExecutors) map[string]string {
 	return results
 }
 
+func contains(list []string, target string) bool {
+	for _, str := range list {
+		if str == target {
+			return true
+		}
+	}
+	return false
+}
+
 // mapTaskLabels returns a map of all task labels prefixed DCOS_METRICS_
-func mapTaskLabels(labels *mesos.Labels) map[string]string {
+// or included in list of whitelisted labels
+func mapTaskLabels(labels *mesos.Labels, whitelist []string) map[string]string {
```

Comment:
I would suggest that we rather implement `whitelist` as a `map[string]bool` so that our lookups (run for **EVERY** label-metric pair) are O(1) instead of O(m).